safemode.c

```diff
@@ -39,85 +43,106 @@ int main(int argc, char** argv)
                 ret = 0;
                 break;
             }
             // break on home button press event
-            if ((read(fd, &ev, sizeof(struct input_event)) != -1)
-                    && (ev.code == 61) && (ev.value == 1)) {
+            if ((read(fd, &ev, sizeof(struct input_event)) != -1) && (ev.code == 61) && (ev.value == 1)) {
                 ret = 1;
                 break;
             }
             count++;
-            usleep(1000); // microseconds
+            usleep(1000);    // microseconds
         }
     }
 
     if (ret == 1) {
         // start usb gadget
-        fprintf(stdout, "starting %s mode, press the button to stop it\n", argv[1]);
-        fb.row = 0;
-        fb.halign = CENTER;
-        fb.valign = CENTER;
-        fb.is_cleared = true;
+        fprintf(stdout, "Starting %s mode, press the button to stop it\n", argv[1]);
+        fb.row         = 0;
+        fb.halign      = CENTER;
+        fb.valign      = CENTER;
+        fb.is_cleared  = true;
         fb.is_flashing = true;
         if (mode == usbnet) {
-            fbink_print_image(FBFD_AUTO, "/usr/share/safemode/images/usbnet.png", 0, 0, &fb);
-            system("/usr/share/safemode/scripts/enable-usbnet.sh\n");
+            fbink_print_image(fbfd, "/usr/share/safemode/images/usbnet.png", 0, 0, &fb);
+            system("/usr/share/safemode/scripts/enable-usbnet.sh");
         } else if (mode == usbms) {
-            fbink_print_image(FBFD_AUTO, "/usr/share/safemode/images/usbms.png", 0, 0, &fb);
-            system("/usr/share/safemode/scripts/enable-usbms.sh\n");
+            fbink_print_image(fbfd, "/usr/share/safemode/images/usbms.png", 0, 0, &fb);
+            system("/usr/share/safemode/scripts/enable-usbms.sh");
         }
 
         // wait for another home button press event, this time without timeout.
         while (1) {
             usleep(1000);
             // break on home button press event
-            if ((read(fd, &ev, sizeof(struct input_event)) != -1)
-                    && (ev.code == 61) && (ev.value == 1)) {
+            if ((read(fd, &ev, sizeof(struct input_event)) != -1) && (ev.code == 61) && (ev.value == 1)) {
                 break;
             }
         }
         // stop usb gadget
-        fprintf(stdout, "button pressed, stopping %s mode\n", argv[1]);
+        fprintf(stdout, "Button pressed, stopping %s mode\n", argv[1]);
         if (mode == usbnet) {
-            system("/usr/share/safemode/scripts/disable-usbnet.sh\n");
+            system("/usr/share/safemode/scripts/disable-usbnet.sh");
         } else if (mode == usbms) {
-            system("/usr/share/safemode/scripts/disable-usbms.sh\n");
+            system("/usr/share/safemode/scripts/disable-usbms.sh");
         }
-        close(fd);
-        fbink_printf(FBFD_AUTO, NULL, &fb, " ");
-        exit(EXIT_SUCCESS);
+        fbink_print(fbfd, " ", &fb);    // FIXME: Not sure what the intent is here? fbink_cls, maybe?
```

Comment:
The intent was cleaning the dots without changing the background image. Not sure if that can be done with fbink_cls.

Reply:
Oh, then, yeah, no, fbink_cls would do a full white screen.